prometric-derive/src/expand.rs

```diff
@@ -26,9 +26,10 @@ pub(super) struct MetricsAttr {
 }
 
 enum MetricType {
-    Counter(Ident, Type),
-    Gauge(Ident, Type),
-    Histogram(Ident),
+    Counter(TypePath, Type),
+    Gauge(TypePath, Type),
+    Histogram(TypePath),
+    Summary(TypePath),
 }
```

Comment:
This could benefit from a little bit of documentation regarding expected usage in library code, for example:

```rs
/// A wrapper over [`prometric`] metric types, containing their type path and generic
/// arguments, if any.
///
/// ```
/// use syn::parse_str;
///
/// let counter_metric_type =
///     MetricType::from_path(parse_str("::prometric::Counter<u64>").unwrap()).unwrap();
/// let guauge_metric_type =
///     MetricType::from_path(parse_str("Gauge").unwrap()).unwrap();
/// ```
enum MetricType {
    Counter(TypePath, Type),
    Gauge(TypePath, Type),
    Histogram(TypePath),
    Summary(TypePath),
}
```